src/chainparams.cpp

```diff
@@ -240,11 +240,13 @@ static CTestNetParams testNetParams;
 /**
  * Regression test
  */
+const int DEFAULT_REGTEST_HALVING_INTERVAL = 150;
+
 class CRegTestParams : public CChainParams {
 public:
     CRegTestParams() {
         strNetworkID = "regtest";
-        consensus.nSubsidyHalvingInterval = 150;
+        consensus.nSubsidyHalvingInterval = DEFAULT_REGTEST_HALVING_INTERVAL;
```

Comment:
Yes, that was the initial solution I have opted for, but then I discovered that `CRegTestParams` is constructed before `ReadConfigFile` is called and the hasmaps `GetArg` read were empty, so it returned always the `DEFAULT_REGTEST_HALVING_INTERVAL`. I noticed that few lines below in https://github.com/Conio/bitcoin/blob/37584b2191d1d7bc8cc39822b488abd989780994/src/chainparams.cpp#L344-L347 was used this function to update bip39 parameters so I decided to go for a similar solution.